packages/cli/src/lib/autorun/command-object.ts

```diff
@@ -0,0 +1,19 @@
+import { ArgumentsCamelCase, CommandModule } from 'yargs';
+import { collectAndPersistReports, upload } from '@code-pushup/core';
+import { uploadConfigSchema } from '@code-pushup/models';
+import { CommandBase } from '../implementation/model';
+
+export function yargsAutorunCommandObject() {
+  return {
+    command: 'autorun',
+    describe: 'Autorun executes the collect and upload command after another',
+    handler: async <T>(args: ArgumentsCamelCase<T> & CommandBase) => {
+      const options = {
+        ...args,
+        upload: uploadConfigSchema.parse(args.upload),
```

Comment:
I don't understand why the `upload` part of the config is parsed within the handler, unlike everything else. Doesn't the middleware take care of this? Isn't this parsing it again?

Reply:
We have `upload` as optional in `CoreConfig` and the schema

Reply:
In the case of `autorun` (unlike `upload` command), I would say if the upload config is missing, then we just don't run the `upload` command, only `collect` (we can print a warning to the console instead). I believe this is more inline with the Lighthouse CI `autorun`.

Reply:
I like the addition to `autorun`. This might help to not loose data.

Reply:
But then we can't point portal-less users to `code-pushup` or `code-pushup autorun`, because it'll fail. And I'm not sure `autorun` should be the default command in that case, because it won't work out of the box.

Wouldn't it be better to print something like this to the end of stdout?
"Your report is ready, but hasn't been uploaded to the portal. If you add upload credentials to the config file, you can then run `code-pushup upload`."

Reply:
Might've misunderstood if you were agreeing with me or not, now that I see the code changes :sweat_smile:

Reply:
I was arranging.. I console.warned. 
But like to think more about "work out of the box".

> `autorun` collected that report, but hasn't been uploaded to the portal because of missing credentials. Add upload credentials to the config file, and run `code-pushup upload` to upload the collected report.